mono/profiler/Makefile.am

```diff
@@ -111,39 +112,51 @@ endif
 # the shared version on those targets since it's useful for e.g. simulator
 # builds.
 
+CFLAGS := $(filter-out @CXX_REMOVE_CFLAGS@, @CFLAGS@)
+
 libmono_profiler_aot_la_SOURCES = aot.c
+libmono_profiler_aot_la_CFLAGS = @CXX_ADD_CFLAGS@
 libmono_profiler_aot_la_LIBADD =  $(libmono_dep) $(glib_libs)
 libmono_profiler_aot_la_LDFLAGS = $(prof_ldflags)
 libmono_profiler_aot_static_la_SOURCES = aot.c
+libmono_profiler_aot_static_la_CFLAGS = @CXX_ADD_CFLAGS@
 libmono_profiler_aot_static_la_LDFLAGS = -static
 
 libmono_profiler_log_la_SOURCES = log.c log-args.c
+libmono_profiler_log_la_CFLAGS = @CXX_ADD_CFLAGS@
 libmono_profiler_log_la_LIBADD = $(libmono_dep) $(glib_libs) $(zlib_dep)
 libmono_profiler_log_la_LDFLAGS = $(prof_ldflags)
 libmono_profiler_log_static_la_SOURCES = log.c log-args.c
 libmono_profiler_log_static_la_LDFLAGS = -static
 
 libmono_profiler_coverage_la_SOURCES = coverage.c
+libmono_profiler_coverage_la_CFLAGS = @CXX_ADD_CFLAGS@
 libmono_profiler_coverage_la_LIBADD = $(libmono_dep) $(glib_libs)
 libmono_profiler_coverage_la_LDFLAGS = $(prof_ldflags)
 libmono_profiler_coverage_static_la_SOURCES = coverage.c
+libmono_profiler_coverage_static_la_CFLAGS = @CXX_ADD_CFLAGS@
 libmono_profiler_coverage_static_la_LDFLAGS = -static
 
 if HAVE_VTUNE
 libmono_profiler_vtune_la_SOURCES = vtune.c
 libmono_profiler_vtune_la_CFLAGS = $(VTUNE_CFLAGS)
+libmono_profiler_vtune_la_CFLAGS += @CXX_ADD_CFLAGS@
 libmono_profiler_vtune_la_LIBADD = $(VTUNE_LIBS) $(libmono_dep) $(glib_libs)
 libmono_profiler_vtune_la_LDFLAGS = $(prof_ldflags)
 libmono_profiler_vtune_static_la_SOURCES = vtune.c
 libmono_profiler_vtune_static_la_LDFLAGS = -static
 libmono_profiler_vtune_static_la_CFLAGS = $(VTUNE_CFLAGS)
+libmono_profiler_vtune_static_la_CFLAGS += @CXX_ADD_CFLAGS@
 libmono_profiler_vtune_static_la_LIBADD = $(VTUNE_LIBS)
 endif
 
-mprof_report_SOURCES = mprof-report.c
-mprof_report_LDADD = $(glib_libs) $(zlib_dep)
+mprof_report_SOURCES =
+mprof_report_LDADD = libmprof_report_a-mprof-report.$(OBJEXT) $(glib_libs) $(zlib_dep)
 mprof_report_LDFLAGS = -no-undefined
 
+libmprof_report_a_SOURCES = mprof-report.c
+libmprof_report_a_CFLAGS = @CXX_ADD_CFLAGS@
```

Comment:
It's not immediately obvious why this needs to be a static library just to change its `CFLAGS`.

Reply:
Is this explained in some other place in our makefiles? If so, a comment here pointing to that explanation would be good.